client/src/pages/bookingPage.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC, useState } from "react";
+import { Button, Grid } from "@mui/material";
+import { TimeSetter } from "../components/TimeSetter";
+import { DateSetter } from "../components/DateSetter";
+import { DeskList } from "../components/DeskList";
+import img1 from "../images/layout.jpg";
+
+export const FormContext = React.createContext<any>(null);
+
+const BookingPage: FC = () => {
+  const handleBooking = async () => {
+    await fetch(`${process.env.REACT_APP_ROOT_URL}api/book`, {
+      method: "PATCH",
+      headers: {
+        "Content-type": "application/json; charset=UTF-8",
+      },
+      body: JSON.stringify({
+        bookingId: `booking ${selectedDesk}.${Date.now()}`,
+        startTime:
+          new Date(
```

Comment:
this triggers me because the logic for startTime and endTime is nearly the same. Could you think of a way to deduplicate this?